tests/runner/src/fluvio-integration-derive/src/lib.rs

```diff
@@ -40,40 +40,44 @@ pub fn fluvio_test(args: TokenStream, input: TokenStream) -> TokenStream {
             Err(_err) => panic!("Parse failed"),
         };
 
+    // Serializing to string to pass into quote! block
     let fn_test_reqs_str =
         serde_json::to_string(&fn_test_reqs).expect("Could not serialize test reqs");
 
     // Read the user test
     let fn_user_test = parse_macro_input!(input as ItemFn);
 
     // If test name is given, then use that instead of the test's function name
-    let test_name = if let Some(req_test_name) = &fn_test_reqs.test_name {
+    let out_fn_iden = if let Some(req_test_name) = &fn_test_reqs.test_name {
         Ident::new(&req_test_name.to_string(), Span::call_site())
     } else {
         fn_user_test.sig.ident
     };
 
-    let test_body = &fn_user_test.block;
+    let test_name = out_fn_iden.to_string();
 
-    let out_fn_iden = Ident::new(&test_name.to_string(), Span::call_site());
-    let async_inner_fn_iden = Ident::new(
-        &format!("{}_inner", &test_name.to_string()),
-        Span::call_site(),
-    );
+    // We're going to wrap the the async test into a sync to store fn pointer w/ `inventory` crate
+    let async_inner_fn_iden = Ident::new(&format!("{}_inner", &test_name), Span::call_site());
 
+    // Enforce naming convention for converting dyn TestOption to concrete type
     let test_opt_ident = Ident::new(
         &format!("{}TestOption", &test_name).to_pascal_case(),
         Span::call_site(),
     );
 
-    let out_fn_str = test_name.to_string();
+    // Finally, the test body
+    let test_body = &fn_user_test.block;
 
     let output_fn = quote! {
 
         pub fn validate_subcommand(subcmd: Vec<String>) -> Box<dyn TestOption> {
             Box::new(#test_opt_ident::from_iter(subcmd))
         }
 
+        pub fn requirements() -> TestRequirements {
+            serde_json::from_str(#fn_test_reqs_str).expect("Could not deserialize test reqs")
```

Comment:
Where do the test requirements come from? Are we certain they will always be JSON? Can we avoid this `.expect`?